File: apps/api/AssetManagement.Api/DTOs/SavedViewDtos.cs
namespace AssetManagement.Api.DTOs;

public record SavedViewDto(
    Guid Id,
    string EntityType,
    string Name,
    bool IsDefault,
    string Configuration,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

public record CreateSavedViewRequest(
    string EntityType,
    string Name,
    string Configuration
);

public record UpdateSavedViewRequest(
    string Name,
    string Configuration
);

File: apps/api-kt/src/main/kotlin/com/assetmanagement/api/dto/SavedViewDtos.kt
package com.assetmanagement.api.dto

import java.time.Instant
import java.util.*

data class SavedViewDto(
    val id: UUID,
    val entityType: String,
    val name: String,
    val isDefault: Boolean,
    val configuration: String,
    val createdAt: Instant,
    val updatedAt: Instant
)

data class CreateSavedViewRequest(
    val entityType: String,
    val name: String,
    val configuration: String
)

data class UpdateSavedViewRequest(
    val name: String,
    val configuration: String
)
